### DISCUSSION TITLE
Populate with copies: Build Array "Polar" Offset not working

### DISCUSSION THREAD

**Discussion by wpegley** [DeepSOIC/Lattice2#62] [General]
OS: Windows 10 Version 2009
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 0.19.24267 +99 (Git)
Build type: Release
Branch: Branch_0.19.3
Hash: 6530e364184ce05ccff39501e175cf2237e6ee4b
Python version: 3.8.6+
Qt version: 5.15.2
Coin version: 4.0.1
OCC version: 7.5.3
Locale: English/Australia (en_AU)

The Attached Video & File might confirm a bug, Works fine on full circle - however I've exhausted all options on a semicircle to get the chain to offset. The polar array does move with Offset, however the parts fail to rebuild.

https://youtu.be/KRElDkAlpYc
https://drive.google.com/file/d/1s2V_Wm77bde4UPMKmBb_Uz8H78vEYWfn/view?usp=sharing

**Comment by DeepSOIC** [comment]
it is not a bug. With 'Build Array', the first occurrence is always where the original is.

On a technical level, 'Build Array' uses the first placement of the array as the reference (or handle to move the object). The array itself shifts, but the reference placement shifts too, negating the effect. Array Radius has no effect for the same reason. Use any other populate option if you want offset to have effect.

I have an experimental branch where it is fixed, but it's unfinished.

**Comment by wpegley** [comment]
Understood thx for the quick reply & great work on the WB.
